### ISSUE TITLE
Spectator mode has errors when using 1.17.1 server under build#454

### ISSUE THREAD

**Issue by Chonghua-05** [PaperMC/Velocity#1470]
### Expected Behavior

When using velocity to connect to the backend server, in spectator mode, you can use the spectator mode to pass through blocks and use the inventory to quickly teleport normally.

### Actual Behavior

In spectator mode, you cannot pass through blocks or use the inventory to fast teleport.

### Steps to Reproduce

With velocity build#454, 1.17.1fabric server, and Fabric Loader 0.14.25, use the corresponding client version to enter the server, switch to spectator mode, try to pass through the block, and try to use the inventory to quickly teleport when there are multiple people on the server. These can be reproduced.

### Plugin List

without any plugin

### Velocity Version

velocity 3.4.0 build#454

### Additional Information

I suspect that the game mode data on the server side of the old version (1.17.1) was not forwarded to the client by velocity normally.
In addition, my issue is machine translated, I apologize for the difficulty in reading.

**Comment by Trirrin** [comment]
Yes, I encountered the same problem when using 1.17.1 as the upstream server and also used version build 454. This bug is unrelated to any plugins or mods. The specific problem manifestation is the same as described in this issue: #1315 . However, the cause may be slightly different.When using velocity 3.1.1 (build 98), this situation does not occur.So, I think there is some truth to what @Chonghua-05  said in the Additional Information section.
